### PR TITLE
Add support for multiple Keyword Position formatting

### PR DESCRIPTION
**Pull request by inferrinizzard**
- added support for tenSpace formatting (aligned with central column)
  - tenSpaceLeft (left-aligned keywords)
  - tenSpaceRight (right-aligned keywords)
- could potentially make the width of the columns a top-level arg
- added example markdown file to describe the 3 styles

- added logic for edge case between interaction with tabulateAlias and tenSpace formats
- removed extra indent on inner block started with FROM
- fixed some tests related to ^
- moved ON keyword out of standard Reserved list and into Newline Reserved

### CODE REVIEW

File: src/core/Formatter.ts [hunk 1]
```diff
@@ -8,18 +8,20 @@ import {
...
+	ZWS,
 } from './token';
 import Tokenizer from './Tokenizer';
 import type { FormatOptions } from '../sqlFormatter';
-import { AliasMode, CommaPosition, NewlineMode } from '../types';
+import { AliasMode, CommaPosition, KeywordMode, NewlineMode } from '../types';
 
 export default class Formatter {
-	cfg: FormatOptions;
+	cfg: FormatOptions & { tenSpace?: boolean };
```
Comment (inferrinizzard) [inline src/core/Formatter.ts:24]: unideal place to store this tenSpace marker

File: src/core/Formatter.ts [hunk 2]
```diff
@@ -355,10 +372,16 @@ export default class Formatter {
 
 		query = this.addNewline(query);
 
-		this.indentation.increaseTopLevel();
+		if (this.cfg.tenSpace) {
```
Comment (inferrinizzard) [inline src/core/Formatter.ts:375]: simplify this block eventually
